### PR TITLE
Update `source_gen` and `build_runner`

### PR DESCRIPTION
**Pull request by tim-smart**
Can remove the git dependency when https://github.com/dart-lang/source_gen publishes their null safety work.

### CODE REVIEW

**Comment by rrousselGit** [comment]
source_gen ^1.0.0 was published. Do you want to finish this PR or should I take over?

**Comment by tim-smart** [comment]
@rrousselGit I have updated the constraint. Not sure if the CI tests will be happy, so you can take over from here :)

**Comment by tim-smart** [comment]
Yeah as I thought, the CI environment is having issues with the `test` package:

```
functional_widget:
error: Library 'file:///home/runner/.pub-cache/hosted/pub.dartlang.org/test-1.16.8/bin/test.dart' was compiled with sound null safety (in strong mode) and it requires --sound-null-safety option at runtime
```

**Comment by themadmrj** [comment]
Any ETA on this? I have to use a custom fork just because of this dependency...

**Comment by rrousselGit** [comment]
The blocker is code_builder https://github.com/dart-lang/code_builder/issues/303

File: packages/code_gen_tester/lib/src/tester.dart [hunk 1]
```diff
@@ -71,7 +71,7 @@ class _SourceGenTesterImpl implements SourceGenTester {
       [BuildStep? buildStep]) async {
     final generated =
         await generator.generate(library, buildStep ?? _BuildStepImpl());
-    final output = formatter.format(generated ?? '');
```
Comment (tim-smart) [inline packages/code_gen_tester/lib/src/tester.dart]: `generated` is a `String?`, so this produces an analyzer error?

See https://pub.dev/documentation/source_gen/latest/source_gen/Generator/generate.html

**Comment by fdietze** [comment]
Until this PR is merged, you can already use it like this in your `pubspec.yaml`:

```yaml
dependencies:
  functional_widget_annotation:
    git:
      url: git://github.com/tim-smart/functional_widget.git
      ref: "ad7f47788160137fe17c9038ab0298e89b4a6f34" # https://github.com/rrousselGit/functional_widget/pull/86
      path: packages/functional_widget_annotation

dev_dependencies:
  functional_widget:
    git:
      url: git://github.com/tim-smart/functional_widget.git
      ref: "ad7f47788160137fe17c9038ab0298e89b4a6f34" # https://github.com/rrousselGit/functional_widget/pull/86
      path: packages/functional_widget

dependency_overrides:
  functional_widget_annotation:
    git:
      url: git://github.com/tim-smart/functional_widget.git
      ref: "ad7f47788160137fe17c9038ab0298e89b4a6f34" # https://github.com/rrousselGit/functional_widget/pull/86
      path: packages/functional_widget_annotation
```

**Comment by rrousselGit** [comment]
Looks like code_builder has been released, so I'll finish this

**Pull request merged**